### PR TITLE
Clean up WASM_BACKEND usage in emcc.py

### PR DESCRIPTION
**Pull request by kripken**
See #11860 

Btw, just to confirm, `add_dylink_section` seems no longer used - I guess the
wasm backend emits that section for us? (do we not need to update it?)

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -1372,61 +1371,6 @@ def add_emscripten_metadata(js_file, wasm_file):
       f.write(contents)
       f.write(orig[8:])
 
-  @staticmethod
-  def add_dylink_section(wasm_file, needed_dynlibs):
```
Comment (sbc100) [inline tools/shared.py:1376]: I have a PR in progress that modifies this method to work with the wasm backend.   So can you maybe leave it in for now?

File: emcc.py [hunk 2]
```diff
@@ -2876,12 +2723,6 @@ def do_binaryen(target, asm_target, options, memfile, wasm_binary_target,
     building.save_intermediate(wasm_binary_target, 'pre-ctors.wasm')
     building.eval_ctors(final, wasm_binary_target, binaryen_bin, debug_info=intermediate_debug_info)
 
-  # after generating the wasm, do some final operations
-  if shared.Settings.SIDE_MODULE and not shared.Settings.WASM_BACKEND:
```
Comment (sbc100) [inline emcc.py:2880]: Could you leave this in but maybe commented or with a TODO?

File: emcc.py [hunk 3]
```diff
@@ -2876,12 +2723,6 @@ def do_binaryen(target, asm_target, options, memfile, wasm_binary_target,
...
-    shared.WebAssembly.add_dylink_section(wasm_binary_target, shared.Settings.RUNTIME_LINKED_LIBS)
-    if not DEBUG:
-      os.unlink(asm_target) # we don't need the asm.js, it can just confuse
```
Comment (sbc100) [inline emcc.py:2883]: You can remove these last two lines though.

File: emcc.py [hunk 4]
```diff
@@ -1195,20 +1192,14 @@ def get_last_setting_change(setting):
...
       # as there is no JS. an exception are side modules, as we can't tell at
       # compile time whether JS will be involved or not - the main module may
       # have JS, and the side module is expected to link against that.
       # we also do not support standalone mode in fastcomp.
       shared.Settings.STANDALONE_WASM = 1
 
-    # We allow this error to be supressed by the environment so that we can run the test
-    # suite against fastcomp for the time being.
-    # See: https://github.com/emscripten-core/emscripten/issues/11319
-    if not shared.Settings.WASM_BACKEND and 'EMCC_ALLOW_FASTCOMP' not in os.environ:
```
Comment (sbc100) [inline emcc.py:1209]: We still want this right?

File: tools/shared.py (hunk 1 above)
Comment (kripken) [reply tools/shared.py:1376]: Ok, I'll restore `add_dylink_section`.

File: emcc.py (hunk 4 above)
Comment (kripken) [reply emcc.py:1209]: Hmm, why would we want to look at `EMCC_ALLOW_FASTCOMP`? Fastcomp won't work with all the removed code anyhow, and we aren't running those tests.

Or do you mean we want an error for trying to use fastcomp? I don't think it's possible to get to this code location. We error anyhow on trying to set `WASM_BACKEND` - we always have, before this PR, it's autodetected. And we've even removed that autodetection code?

File: emcc.py (hunk 2 above)
Comment (kripken) [reply emcc.py:2880]: Sure, done.

File: emcc.py (hunk 4 above)
Comment (sbc100) [reply emcc.py:1209]: I guess I mean we still need the error case... but you can drop the EMCC_ALLOW_FASTCOMP part

**Pull request merged**